workers/unity/Packages/com.improbable.gdk.playerlifecycle/PlayerLifecycleHelper.cs

```diff
@@ -90,9 +82,15 @@ public static bool IsOwningWorker(SpatialEntityId entityId, World workerWorld)
             return worker.Connection.GetWorkerId() == ownerId;
         }
 
-        public static void AddClientSystems(World world)
+        public static void AddClientSystems(World world, bool autoRequestPlayerCreation = true,
+            Vector3 spawnPosition = default,
+            byte[] serializedArguments = null)
```

Comment:
Can we assume that we will know the arguments at this time? (Immediately after the worker connects)

Reply:
There could be some default arguments that a user may set once and never use again that are known at this point

Reply:
Hmmm - I'm not sure what the best API here is, especially since this splits the API across different places. 

Use cases:
- Automatic player spawning with preset arguments
- Automatic player spawning with dynamic (Is this a valid use case? This would require the user to ensure that they set the arguments _before_ the player lifecycle systems ran. In which case why do not a manual player spawn after you set the arguments)
- Manual player spawning with preset arguments
- Manual player spawning with dynamic arguments

Reply:
discussed offline:
* auto-spawning will not send any serialized arguments
* `RequestPlayerCreation` will take the serialized arguments directly (instead of having a different place to set it)
* keep the private `byte[]` field to cache the requested arguments for retries